pandas/core/arrays/datetimelike.py

```diff
@@ -864,6 +1264,51 @@ def _evaluate_compare(self, other, op):
         result[mask] = filler
         return result
 
+    def _reduce(self, name, skipna=True, **kwargs):
+        op = getattr(self, name, None)
+        if op:
+            return op(skipna=skipna)
+        else:
+            return super()._reduce(name, skipna, **kwargs)
+
+    # --------------------------------------------------------------
+    # Reductions
+
+    def any(self, skipna=True):
+        if skipna:
+            values = self[~self.isnan]
+        else:
+            values = self
+
+        # TODO: Should any period be considered Falsey?
+        return len(values)
```

Comment:
`len(values) > 0`?

Reply:
I'm hoping that https://github.com/pandas-dev/pandas/pull/23890 is mergeable first, and I can wipe out my reduction stuff.